bin/workflows/pycbc_make_coinc_search_workflow

```diff
@@ -155,7 +155,23 @@ precalc_psd_files = wf.setup_psd_workflow(workflow, analyzable_segs,
 bank_files = wf.setup_tmpltbank_workflow(workflow, analyzable_segs,
                                          datafind_files, output_dir="bank",
                                          psd_files=precalc_psd_files)
-hdfbank = wf.convert_bank_to_hdf(workflow, bank_files, "bank")
+
+# Determine the extension of the bank file taken as input
+cp = workflow.cp
+tmpltbankMethod = cp.get_opt_tags("workflow-tmpltbank", "tmpltbank-method", '')
+if tmpltbankMethod == "PREGENERATED_BANK":
+    pre_gen_bank = cp.get_opt_tags('workflow-tmpltbank',
+                                           'tmpltbank-pregenerated-bank', '')
+    bank_filename, ext = (pre_gen_bank.split('/')[-1].split('.'))
+
+# If the bank is a hdf or h5 file do not change the format. If not,
+# convert it to hdf
+if ext == 'hdf' or ext == 'h5' :
+    hdfbank = bank_files
+else :
+    hdfbank = wf.convert_bank_to_hdf(workflow, bank_files, "bank")
+logging.info("Input bank is a %s file", ext)
```

Comment:
Do you think it's possible to move all this into setup_tmpltbank_workflow? Maybe add a return_format key-word argument. If not given the bank just comes back in whatever format it is supplied in, if hdf, it does what you've added here, if XML it either returns an XML file (if we have one) or fails if we have something else. If value is given and is *not* XML or HDF then fail. This then allows this to be easily used in other applications.